services/src/main/scala/cromwell/services/metadata/WorkflowProcessingEventPublishing.scala

```diff
@@ -0,0 +1,38 @@
+package cromwell.services.metadata
+
+import java.time.OffsetDateTime
+
+import akka.actor.ActorRef
+import common.util.VersionUtil
+import cromwell.core.WorkflowId
+import cromwell.core.WorkflowProcessingEvents.EventKey._
+import cromwell.core.WorkflowProcessingEvents._
+import cromwell.services.metadata.MetadataService.PutMetadataAction
+
+import scala.util.Random
+
+object WorkflowProcessingEventPublishing {
```

Comment:
Still reviewing, but should this go in `engine`? It seems to a) only be used by engine, and b) is the **only** reason `services` now has to depend on `core`.